internal/service/certificatesigningrequest.go

```diff
@@ -19,26 +19,27 @@ import (
 
 const DefaultEnrollmentCertExpirySeconds int32 = 60 * 60 * 24 * 7 // 7 days
 
-func signApprovedCertificateSigningRequest(ca *crypto.CA, request *api.CertificateSigningRequest) error {
+func signApprovedCertificateSigningRequest(ca *crypto.CA, request *api.CertificateSigningRequest) ([]byte, error) {
 	if request == nil {
-		return errors.New("signApprovedCertificateSigningRequest: CertificateSigningRequest is nil")
+		return nil, errors.New("signApprovedCertificateSigningRequest: CertificateSigningRequest is nil")
 	}
 
-	// NOTE: this currently only requires one approval; this will change in the future based on policy
-	if api.IsStatusConditionTrue(request.Status.Conditions, api.CertificateSigningRequestDenied) {
-		return fmt.Errorf("signApprovedCertificateSigningRequest: certificate signing request has been denied and cannot be signed")
-	}
-	if !api.IsStatusConditionTrue(request.Status.Conditions, api.CertificateSigningRequestApproved) {
-		return fmt.Errorf("signApprovedCertificateSigningRequest: certificate signing request is not approved and cannot be signed")
+	// TODO - we don't check for Approved status here anymore as it makes more sense to set Approved only after a certiicate
+	// is updated - otherwise a CSR may be marked as Approved but not actually make it through the entire Approval process (ex.
+	// certificate may be missing). this assumes the checks for "should be approved/signed" happen in the calling function
+	if request.Status.Conditions != nil && len(request.Status.Conditions) > 0 {
+		if api.IsStatusConditionTrue(request.Status.Conditions, api.CertificateSigningRequestDenied) {
```

Comment:
If the length of conditions is greater than 0 - does it mean there should always be a `CertificateSigningRequestDenied` condition ?

Reply:
This check isn't needed - request.Status.Conditions can't be nil, and we don't need to check its length.  BTW, we should probably set both the approved and denied conditions to false when creating the CSR, right?

Reply:
I think it could also be another approval, or something else - we aren't really constraining it AFAIA

Reply:
Failing to check the len was resulting in errors

Reply:
But I think we are removing these checks anyway

Reply:
Yeah what Avishay said ^

> BTW, we should probably set both the approved and denied conditions to false when creating the CSR, right?
I think this is happening but I will double check

Reply:
Creation of the CSR sets empty Conditions: https://github.com/flightctl/flightctl/blob/main/internal/store/model/certificatesigningrequest.go#L41 - is that good enough, or do they need to explicitly exist and be marked as "false"?

Reply:
Explicitly marked as "false" - otherwise the default is "unknown"

Reply:
Okay, can do in an upcoming PR